scripts/webhook-migration/updateWebhooks.sh

```diff
@@ -0,0 +1,182 @@
+[ -z "${AUTH_URL}" ] && echo "AUTH_URL needs to be set" && exit 1
+[ -z "${OPENSHIFT_URL}" ] && echo "OPENSHIFT_URL needs to be set" && exit 1
+[ -z "${OPENSHIFT_API_TOKEN}" ] && echo "OPENSHIFT_API_TOKEN needs to be set" && exit 1
+[ -z "${AUTH_TOKEN}" ] && echo "AUTH_TOKEN needs to be set" && exit 1
+[ -z "${JENKINS_OLD_URL_SUFFIX}" ] && echo "JENKINS_OLD_URL_SUFFIX needs to be set" && exit 1
+[ -z "${JENKINS_PROXY_URL}" ] && echo "JENKINS_PROXY_URL needs to be set" && exit 1
+
+# Ensure required tools are installed
+hash jq 2>/dev/null || { echo >&2 "'jq' is required."; exit 1; }
+hash curl 2>/dev/null || { echo >&2 "'curl' is required."; exit 1; }
+hash oc 2>/dev/null || { echo >&2 "'oc' is required."; exit 1; }
+hash osio 2>/dev/null || { echo >&2 "'osio' is required."; exit 1; }
+
+###############################################################################
+# Processes the build configuration of
+# the OpenShift cluster and extracts GitHub
+# repo URLs which might need updating.
+# Globals:
+#   OPENSHIFT_URL
+#   OPENSHIFT_API_TOKEN
+#   NAME_SPACES
+#   REPOS
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+findGitHubRepos() {
+  oc login ${OPENSHIFT_URL} --token=${OPENSHIFT_API_TOKEN} > /dev/null
+
+  #local projects=$(oc projects -q)
+  local projects=( "hferents" )
+  for project in ${projects}; do
+   oc project ${project} > /dev/null;
+   local tmp=($(oc get bc -o json | jq -r ' .items[] | select(.spec.source.git.uri) | .metadata.namespace + " " + .spec.source.git.uri[19:-4]'))
+   if [ ${#tmp[@]} -eq 0  ]; then
+    continue
+   fi
+   NAME_SPACES+=(${tmp[0]})
+   REPOS+=(${tmp[1]})
+  done
+}
+
+###############################################################################
+# Finds the UUIDs for the discovered namespaces
+# Globals:
+#   NAME_SPACES - Namespaces to iterate
+#   UUIDS       - Array which gets populated with UUIDs
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+findUUIDs() {
+  for i in "${!NAME_SPACES[@]}"; do
+    uuid=$(curl -sgSL -H "Authorization: Bearer ${AUTH_TOKEN}" ${TENTANT_URL}/api/tenants?namespace=${NAME_SPACES[$i]}\&master_url=${OPENSHIFT_URL} | jq -r .data[0].id)
+    [ -z "${uuid}" ] && echo "WARN: Unable to determine UUID for ${NAME_SPACES[$i]}" >&2
+    UUIDS[$i]=${uuid}
+  done
+}
+
+###############################################################################
+# For each UUID get an OSIO token
+# Globals:
+#   UUIDS      - User ids to iterate
+#   OSIO_TOKEN - Array which gets populated with OSIO tokens
+# Arguments:
+#   None
+# Returns:
+#   None
+###############################################################################
+getOsioTokens() {
+  for i in "${!UUIDS[@]}"; do
+    # TODO Issue #94 - How to pass the key? ATM it is a file path? Should it be read from env?
+    osio_token=$(echo "osio token -k keyfile -u ${UUIDS[$i]}")
```

Comment:
ATM the `osio` CLI expects a file path to the private key. I guess for the in-container mode, we read the key also out of the environment? Or would it be mounted?

Reply:
The easiest would be to get it from env var

Reply:
That's what I  I thought. Leet me make a change then